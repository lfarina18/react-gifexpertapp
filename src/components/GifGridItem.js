import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {


    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

/*
    1. Enzime
    2. Enzyme to Json
    3. Debe de mostrar el componente
        * shallow
        * wrapper
        * wrapper .toMatchSnapshot
*/

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}