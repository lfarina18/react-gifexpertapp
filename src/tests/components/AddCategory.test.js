import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from "enzyme"
import { AddCategory } from '../../components/AddCategory'


describe('Pruebas sobre el componente <AddCategory/>', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);

    });

    test('debe de mostrar el componente correctamente', () => {


        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);
    });


    test('no debe de postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();

    });

    test('debe de llamar setCategories y limpiar la caja de texto', () => {

        const value = 'Hola Mundo';

        //1. simular el inputChange
        wrapper.find('input').simulate('change', { target: { value } });

        //2. simular el submit
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        //3. setCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        //4. El valor del input debe estar ''

        expect(wrapper.find('input').prop('value')).toBe('');

    })








})
