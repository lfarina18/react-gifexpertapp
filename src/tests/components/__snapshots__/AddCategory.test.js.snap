// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <AddCategory/> debe de mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
